File: 프로그래머스/unrated/181895. 배열 만들기 3/배열 만들기 3.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> intervals) {
    vector<int> x (arr.begin() + intervals[0][0], arr.begin() + intervals[0][1] + 1);
    vector<int> y (arr.begin() + intervals[1][0], arr.begin() + intervals[1][1] + 1);
    
    x.insert(x.end(), y.begin(), y.end());
    
    return x;
}

File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.java
class Solution {
    public int[] solution(int[] arr, int[][] intervals) {
        int[] ints = new int[intervals[0][1] - intervals[0][0] + 1 + intervals[1][1] - intervals[1][0] + 1];
        int idx = 0;
        for (int[] interval : intervals) {
            for (int i = interval[0]; i <= interval[1]; i++) {
                ints[idx++] = arr[i];
            }
        }
        return ints;
    }
}
